db/bigtable_eth1.go

```diff
@@ -1772,6 +1772,28 @@ func (bigtable *Bigtable) GetAddressesNamesArMetadata(names *map[string]string,
 	return *names, outputMetadata, nil
 }
 
+func (bigtable *Bigtable) GetIndexedEth1Transaction(txHash []byte) (*types.Eth1TransactionIndexed, error) {
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(time.Second*30))
+	defer cancel()
+	key := fmt.Sprintf("%s:TX:%x", bigtable.chainId, txHash)
+	row, err := bigtable.tableData.ReadRow(ctx, key)
+
+	if err != nil {
+		return nil, err
+	}
+	if row == nil {
+		return nil, fmt.Errorf("tx not found %x", txHash)
```

Comment:
Why not `return nil, nil` like done, for example, here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/95278d95a21a3c493da5d9c078518bb89b356a76/db/bigtable_eth1.go#L2784

This would make it easier for you to handle the return value.

I honestly would not see this as an error.

Reply:
Yea I thought the caller should decide what to do ... but yea let's keep it simple ;-)